File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let pairs = {"[":"]", "{":"}","(":")"}
    let stack =[]
    for (let c of s){
        if(pairs[c]){
           stack.push(pairs[c])  
        }
        else {
            let item = stack.pop()
            if(item !== c) return false
        }
    }
     return stack.length == 0
};

File: 0020-valid-parentheses/0020-valid-parentheses.rs

impl Solution {
    pub fn is_valid(s: String) -> bool {
        let map = std::collections::HashMap::from([('[', ']'), ('{', '}'), ('(', ')')]);
        let mut stack = Vec::new();

        for c in s.chars() {
            let item = map.get(&c);
            if item != None {
                stack.push(item.unwrap());
                continue;
            }
            let item = stack.pop().unwrap_or(&'*');
            if item != &c {
                return false;
            }
        }

        stack.len() == 0
    }
}